src/core/server/saved_objects/migrations/actions/integration_tests/actions.test.ts

```diff
@@ -366,6 +370,42 @@ describe('migration actions', () => {
       const yellowStatusResponse = await client.cluster.health({ index: 'red_then_yellow_index' });
       expect(yellowStatusResponse.status).toBe('yellow');
     });
+    it('resolves left with "index_not_yellow_timeout" after waiting for an index status to be yellow timeout', async () => {
+      // Create a red index
+      await client.indices
+        .create({
+          index: 'red_then_yellow_index',
+          timeout: '5s',
+          body: {
+            mappings: { properties: {} },
+            settings: {
+              // Allocate 1 replica so that this index stays yellow
+              number_of_replicas: '1',
+              // Disable all shard allocation so that the index status is red
+              index: { routing: { allocation: { enable: 'none' } } },
+            },
+          },
+        })
+        .catch((e) => {});
+
+      // Try to clone into an existing index to ensure that we reach a timeout waiting for index status yellow
+      const cloneIndexPromise = cloneIndex({
```

Comment:
We already have a test for the `cloneIndex` action with this scenario. I think it makes sense to add this new test, but we should call `waitForIndexStatusYellow` directly instead of implicitly relying on cloneIndex calling it.

Reply:
I agree, here's the change:https://github.com/elastic/kibana/pull/130352/files#diff-12a441f56b363a502c5018a1ef62d6310074200507970dedd02ae234121cfeb1R392